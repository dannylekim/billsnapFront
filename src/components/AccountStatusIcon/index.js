import AccountStatusIcon from "./AccountStatusIcon";

export default AccountStatusIcon;