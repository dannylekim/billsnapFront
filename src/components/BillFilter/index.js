import BillFilter from "./BillFilter.jsx";

export default BillFilter;
