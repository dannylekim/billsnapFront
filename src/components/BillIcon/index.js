import BillIcon from "./BillIcon";

export default BillIcon;