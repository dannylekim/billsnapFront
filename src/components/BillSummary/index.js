import BillSummary from './BillSummary';
import {connect} from "react-redux";
import { getTotalAmountOwe, getCountOfBills } from "../../redux/selectors/billsSelectors";


const mapStateToProps = (state) => ({
    billCount: getCountOfBills(state),
    amountToOwe: getTotalAmountOwe(state),
    activeBill: state.bills.activeBill,
  });

export default connect(mapStateToProps)(BillSummary);