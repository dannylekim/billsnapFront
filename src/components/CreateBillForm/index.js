import CreateBillFormContainer from "./CreateBillFormContainer";
import {connect} from "react-redux";
import {createNewBill} from "../../redux/actions/billActions";

const mapDispatchToProps = (dispatch) => ({
  createNewBill: (createBillParam) => dispatch(createNewBill(createBillParam)),
});

export default connect(null, mapDispatchToProps)(CreateBillFormContainer);
