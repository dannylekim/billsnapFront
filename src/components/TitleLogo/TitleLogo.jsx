import React from "react";
import "./styles.scss";

export const TitleLogo = () => {
  return (
    <div className="title__row">
      <div className="title__title">
        <h1>Billsnap</h1>
      </div>
      <div className="title__logo">
        <img
          src="http://via.placeholder.com/150x100.png?text=150x100+Logo"
          alt="Logo"
        />
      </div>
    </div>
  );
};

export default () => {
  return (
    <div className="title">
      <TitleLogo />
      <div className="title__row title__text">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Atque
          dignissimos eos error quo, repellendus repudiandae sunt. Amet
          asperiores atque commodi, deleniti dicta expedita ipsa modi nemo, quos
          rem voluptate voluptates.
        </p>
      </div>
    </div>
  );
};
