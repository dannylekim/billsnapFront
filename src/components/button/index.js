import * as BillsnapButtons from './button.jsx';

console.log(BillsnapButtons);
export BillsnapButtons;