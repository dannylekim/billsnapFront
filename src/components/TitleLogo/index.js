import TitleLogo from "./TitleLogo";
export * from './TitleLogo';
export default TitleLogo;