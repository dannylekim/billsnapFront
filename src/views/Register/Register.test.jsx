import React from "react";
import ReactDOM from "react-dom";
import RegisterPage, { RegisterContainer } from "./Register.jsx";
import { TitleLogo } from "../../components/TitleLogo";

describe("RegisterPage", () => {
  let wrapper;
  let handleMockFunction;

  beforeEach(() => {
    handleMockFunction = jest.fn();
    wrapper = shallow(<RegisterPage />);
  });

  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("RegisterPage should match snap shot", () => {
        matches(<RegisterPage />);
      });
    });

    describe("components", () => {
      // some reason enzyme seems to broken in v3
      test.skip("RegisterPage should contain RegisterContainer and TitleLogo component", () => {
        expect(wrapper.find("RegisterContainer")).to.have.lengthOf(1);
        expect(wrapper.find("TitleLogo").to.have.lengthOf(1));
      });
    });
  });
});

describe("RegisterContainer", () => {
  let wrapper;
  let handleMockFunction;

  beforeEach(() => {
    handleMockFunction = jest.fn();
    wrapper = shallow(
      <RegisterContainer handleButtonClick={handleMockFunction} />
    );
  });

  afterEach(() => {
    handleMockFunction.mockRestore();
  });

  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("RegisterContainer should match snap shot", () => {
        matches(<RegisterContainer />);
      });
    });

    describe("components", () => {});
  });

  describe("functions", () => {});
});
