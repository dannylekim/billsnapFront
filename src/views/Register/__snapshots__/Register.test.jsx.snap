// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterContainer render snapshots ðŸ“¸ RegisterContainer should match snap shot 1`] = `
<div
  className="register__container"
>
  <p>
    Maybe something should go here...
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Register button
  </button>
</div>
`;

exports[`RegisterPage render snapshots ðŸ“¸ RegisterPage should match snap shot 1`] = `
<div
  className="page__register"
>
  <div>
    <div
      className="title"
    >
      <div
        className="title__row"
      >
        <div
          className="title__title"
        >
          Billsnap
        </div>
        <div
          className="title__logo"
        >
          Image
        </div>
      </div>
      <div
        className="title__row title__text"
      >
        <p>
          Placeholder paragraph text Placeholder paragraph text Placeholder paragraph text
        </p>
      </div>
    </div>
  </div>
  <div
    className="register__container"
  >
    <p>
      Maybe something should go here...
    </p>
    <button
      className="btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      Register button
    </button>
  </div>
</div>
`;
